spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -614,18 +613,39 @@ private void processCommits() {
 				}
 				boolean countExceeded = this.count >= this.containerProperties.getAckCount();
 				if (ackMode.equals(AckMode.BATCH) || ackMode.equals(AckMode.COUNT) && countExceeded) {
+					if (this.logger.isDebugEnabled()) {
+						this.logger.debug("Committing in AckMode.COUNT because count " + this.count
+								+ " exceeds configured limit of" + this.containerProperties.getAckCount());
+					}
 					commitIfNecessary();
 					this.count = 0;
 				}
 				else {
 					now = System.currentTimeMillis();
 					boolean elapsed = now - this.last > this.containerProperties.getAckTime();
 					if (ackMode.equals(AckMode.TIME) && elapsed) {
+						if (this.logger.isDebugEnabled()) {
+							this.logger
+									.debug("Committing in AckMode.TIME because time elapsed exceeds configured limit of "
+											+ this.containerProperties.getAckTime());
+						}
 						commitIfNecessary();
 						this.last = now;
 					}
-					else if ((ackMode.equals(AckMode.COUNT_TIME) || this.isManualAck)
-							&& (elapsed || countExceeded)) {
+					else if ((ackMode.equals(AckMode.COUNT_TIME) || this.isManualAck) && (elapsed || countExceeded)) {
+						if (elapsed) {
+							if (this.logger.isDebugEnabled()) {
+								this.logger.debug("Committing in AckMode." + ackMode.name() + " because time elapsed "
+										+ "exceeds configured limit of " + this.containerProperties.getAckTime());
+							}
+							else {
+								if (this.logger.isDebugEnabled()) {
```

Comment:
That makes sense - it could all go under `this.logger.isDebugEnabled()`: it's trying to log different things depending whether the count has been exceeded or the time has elapsed.